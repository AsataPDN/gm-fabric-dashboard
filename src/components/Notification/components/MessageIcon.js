import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";

import Icon from "components/Icon";
import Exclamation from "components/Glyphs/Exclamation";

MessageIcon.propTypes = {
  level: PropTypes.oneOf(["info", "error", "warning", "success"])
};

const MessageIconContainer = styled.div`
  position: absolute;
  left: 0;
  top: 0;
  right: 0;
  bottom: 0;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: flex-end;
  pointer-events: none;
  z-index: 0;

  svg {
    opacity: 0.05;
  }
`;

// Choose an icon to display based on level If Error, show error icon, etc.
function MessageIcon({ level = "info" }) {
  return (
    <MessageIconContainer className="notification-dismiss">
      <Icon
        iconRatio="5"
        borderStyle="BorderTriangleSmall"
        borderOpacity="1"
        borderWidth="2"
      >
        <Exclamation />
      </Icon>
    </MessageIconContainer>
  );
}

export default MessageIcon;
