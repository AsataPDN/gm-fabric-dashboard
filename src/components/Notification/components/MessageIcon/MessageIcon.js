import React from "react";
import PropTypes from "prop-types";

import MessageIconContainer from "./components/MessageIconContainer";

import Icon from "components/Icon";
import Exclamation from "components/Glyphs/Exclamation";

MessageIcon.propTypes = {
  level: PropTypes.oneOf(["info", "error", "warning", "success"])
};

// Choose an icon to display based on level If Error, show error icon, etc.
function MessageIcon({ level = "info" }) {
  return (
    <MessageIconContainer className="notification-dismiss">
      <Icon
        iconRatio="5"
        borderStyle="BorderTriangleSmall"
        borderOpacity="1"
        borderWidth="2"
      >
        <Exclamation />
      </Icon>
    </MessageIconContainer>
  );
}

export default MessageIcon;
