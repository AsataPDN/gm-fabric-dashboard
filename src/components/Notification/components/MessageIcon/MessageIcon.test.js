import React from "react";
import { shallow } from "enzyme";
import MessageIcon from "./MessageIcon";

describe("MessageIcon", () => {
  it("matches snapshot", () => {
    const aMessageIcon = shallow(<MessageIcon />);
    expect(aMessageIcon).toMatchSnapshot();
  });
});
