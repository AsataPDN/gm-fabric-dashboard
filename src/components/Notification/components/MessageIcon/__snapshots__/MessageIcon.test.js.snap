// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageIcon matches snapshot 1`] = `
<styled.div
  className="notification-dismiss"
>
  <Icon
    borderOpacity="1"
    borderStyle="BorderTriangleSmall"
    borderWidth="2"
    iconRatio="5"
  >
    <Exclamation />
  </Icon>
</styled.div>
`;
